deps/Makefile

```diff
@@ -439,8 +463,22 @@ ifeq ($(ARCH), ppc64)
 LLVM_CXXFLAGS += -mminimal-toc
 endif # ARCH == ppc64
 
+# LLVM bug #24157
+ifeq ($(USE_LLVM_SHLIB),1)
+ifeq ($(LLVM_USE_CMAKE),1)
+ifeq ($(LLVM_VER_SHORT),$(filter $(LLVM_VER_SHORT),3.3 3.4 3.5 3.6 3.7))
+$(error USE_LLVM_SHLIB=1 with LLVM_USE_CMAKE=1 requires LLVM > 3.7.x)
```

Comment:
@maleadt which build environments did this error apply to? Could we backport a patch? Then again we might just upgrade to 3.8 very soon after it gets released. I'm testing now to see whether or not this is also an issue on Windows.